mitodl/open-discussions
# try out using Storybook for component development

#### What are the relevant tickets?

none

#### What's this PR do?

This just hacks together a storybook configuration that we can use to develop our components in an isolated way.

It needed a bit of hacking to get it to work. In particular, I had to use our existing webpack configuration for storybook, instead of the one it ships with. Some info about that here: https://storybook.js.org/configurations/custom-webpack-config/

This also just sets up the bare minimum to get it working: a config dir in `static/js/storybook`, a file with a few super basic stories to get us started, and an import for `layout.scss` so our components are styled.

I think there's a lot more we can add, but I thought it might be good to just merge this so people can play around with it and see what they think. I'm going to try using it to simplify our `<a>` and `<Link />` styling a little bit for some upcoming issues.

Note also that it's set as a `devDependency`, this is just because Storybook pulls in a lot of packages, and since we'll never be running this in prod or on a CI server I didn't want to needlessly slow down the build.

#### How should this be manually tested?

check it out! you can try it by running `npm run storybook`. I tried to get it working in the watch docker container, but something was weird with it and it wouldn't connect (even after mapping the port). So I've just been running it locally.
